services/order.py

```diff
@@ -0,0 +1,32 @@
+from db.models import User
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None
+) -> Order:
+    user = User.objects.get(username=username)
+    with transaction.atomic():
+        order = Order.objects.create(user=user,)
+        if date:
+            order.created_at = date
+
+        for ticket in tickets:
+            Ticket.objects.create(movie_session_id=ticket["movie_session"],
+                                  order=order,
+                                  row=ticket["row"],
+                                  seat=ticket["seat"]
+                                  )
+        order.save()

```

Comment:
Don't Repeat Yourself: You call the `save()` method twice, once on line 15 when you create an order, and once on line 25. You should call `save()` only once. To do this, you can use the `update_fields` parameter of the `save()` method. This parameter allows you to specify which fields should be updated. This way, if the `date` is not `None`, you can update the `created_at` field without calling `save()` again.